lib/booking_system.rb

```diff
@@ -0,0 +1,118 @@
+module Hotel
+  class BookingSystem
+    attr_accessor :rooms, :reservations, :all_ids, :blocks
+
+    def initialize
+      @rooms = load_rooms
+      @reservations = []
+      @all_ids = []
+      @blocks = []
+    end
+
+    def load_rooms
+      rooms = []
+      i = 1
+
+      20.times do
+        rooms << i
+        i += 1
+      end
+      return rooms
+    end
+
+    def make_reservation(start_date, end_date)
+      reservation = Hotel::Reservation.new(reservation_id: generate_id, room: assign_available_room(start_date, end_date), start_date: start_date, end_date: end_date, price_per_night: 200)
+      @reservations << reservation
```

Comment:
I love the way you've broken each of these bits of functionality into separate methods! However, it's a little difficult to read all on one line like this. You might split it across multiple lines:

```ruby
reservation = Hotel::Reservation.new(
  reservation_id: generate_id(),
  room: assign_available_room(start_date, end_date),
  start_date: start_date,
  end_date: end_date,
  price_per_night: 200
)
```

I also like adding parentheses to the call to `generate_id()`, even though there are no arguments - this makes it a little clearer that this is a method call.